ContentPatcher/docs/i18n/zh/extensibility.md

```diff
@@ -0,0 +1,303 @@
+
+你可以同时使用基本和进阶API。
+
+## 访问API<a name="access-the-api"></a>
+
+访问API的步骤为：
+
+1. 将Content Patcher设为[`manifest.json`中的依赖](https://stardewvalleywiki.com/Modding:Modder_Guide/APIs/Manifest#Dependencies):
+
+   ```js
+   "Dependencies": [
+      { "UniqueID": "Pathoschild.ContentPatcher", "IsRequired": false }
+   ]
+   ```
+
+2. 把[`IContentPatcherAPI`](../../../IContentPatcherAPI.cs) 复制到你模组里并删除**任何你不需要用的方法，为了兼容未来更改**.
+3. 在你的模组代码中（如[`GameLaunched`事件](https://zh.stardewvalleywiki.com/模组:制作指南/APIs/Events#GameLoop.GameLaunched)中，获取Content Patcher的API：
+   ```c#
+   var api = this.Helper.ModRegistry.GetApi<IContentPatcherAPI>("Pathoschild.ContentPatcher");
+   ```
+4. 使用API延伸Content Patcher功能，（详见以下）。
+
+## 基本API<a name="basic-api"></a>
+### 概念<a name="concepts"></a>
+
+基本API会替你处理大部分设计因素。你只需要考虑以下两点：
+
+<dl>
+<dt>范围</dt>
+<dd>
+
+Content Patcher只有在更新令牌时才会调用你的代码。这可以在存档加载前，加载中，和加载后。如果你的令牌还没准备好，你可以返回null或空列表 这三种情况在 _[添加令牌](#add-a-token)_ 中都照顾到。
+
+</dd>
+
+<dt>数据顺序</dt>
+<dd>
+
+你返回的顺序会影响`valueAt`。推荐使用对于你令牌来说最有意义的顺序，因为内容包作者无法改变此顺序。大部分令牌使用字母数字顺序即可（如 `.OrderBy(p => p, StringComparer.OrdinalIgnoreCase)`）。
+
+</dd>
+</dl>
+
+### 添加令牌<a name="add-a-token"></a>
+
+你可以在`GameLaunched`事件中通过`RegisterToken`添加简单令牌（详见 _[访问API](#access-the-api)_ ）。例如，以下代码创建一个`{{your-mod-id/PlayerName}}`令牌代表当前玩家名：
+
+```c#
+api.RegisterToken(this.ModManifest, "PlayerName", () =>
+{
+    // 存档已加载
+    if (Context.IsWorldReady)
+        return new[] { Game1.player.Name };
+
+    // 存档正在加载
+    if (SaveGame.loaded?.player != null)
+        return new[] { SaveGame.loaded.player.Name };
+
+    // 未加载存档(e.g. 在主页)
+    return null;
+});
+```
+
+`RegisterToken` 有三个参数：
+
+参数   | 类型 | 用途
+---------- | ---- | -------
+`mod`      | `IManifest` | 提供此令牌模组的manifest，你可以直接使用ModEntry的`this.ModManifest`
+`name`     | `string` | 令牌名称，这个名字只需要在你自己模组中独一。Content Patcher会自动给这个名字添加前缀。以上例子的`PlayerName`会变成`your-mod-id/PlayerName`。
+`getValue` | `Func<IEnumerable<string>>` | 一个函数，反馈当前值。如果这反馈null或空列表，这个令牌会以不存在于当前范围来处理，任何使用它的补丁和动态令牌也会被禁用。
+
+现在，任何将你的模组列为依赖的内容包都可以在其字段中使用令牌：
+```js
+{
+   "Format": "2.7.0",
+   "Changes": [
+      {
+         "Action": "EditData",
+         "Target": "Characters/Dialogue/Abigail",
+         "Entries": {
+            "Mon": "Oh hey {{your-mod-id/PlayerName}}! Taking a break from work?"
+         }
+      }
+   ]
+}
+```
+
+## 进阶API<a name="advanced-api"></a>
+### 注意事项<a name="caveats"></a>
+
+大部分时候，强烈推荐使用以上的 _基本API_，因为Content Patcher会帮你处理上下文更新和更改追踪等细节，这方便排查错误，而且无大型更新时绝对不会失效。
+
+如果你真的非常需要更多控制，你可以使用进阶API添加令牌（基本和Content Patcher自带的令牌一致），但是：
+
+* <strong>这是实验性功能。无法保证未来版本会倒退兼容，或在更改之前会收到任何警告。</strong>
+* <strong>这是底层功能。与以上的基本API不同，你必须考虑下面记录的令牌设计注意事项。</strong>
+
+### 概念<a name="concepts-1"></a>
+
+当通过进阶API添加令牌时，你需要考虑这些因素来避免问题：
+
+<dl>
+<dt>范围和数据顺序</dt>
+<dd>
+
+详见 [_基本API：概念_](#concepts)。
+
+</dd>
+
+<dt>上下文更新</dt>
+<dd>
+
+令牌值可以视为缓存的游戏状态，在特定时刻更新（如每天开始）。所有令牌的集合为“上下文”；而“上下文更新”是Content Patcher刷新令牌，生成缓存，检查条件，重加载素材，等等。
+
+**令牌值不能在`UpdateContext`以外更改**。这样做可能会导致贴图错误或游戏彻底崩溃。
```

Comment:
It might be worth keeping the "_severe and undocumented effects_" bit; the examples given (texture errors and game crashes) are only two possible effects, since it could lead to all sorts of other issues.